pkg/cloudprovider/provider/hetzner/provider.go

```diff
@@ -260,6 +277,20 @@ func (p *provider) Create(machine *v1alpha1.Machine, _ *cloudprovidertypes.Provi
 		}
 	}
 
+	if len(c.Firewalls) != 0 {
+		serverCreateOpts.Firewalls = []*hcloud.ServerCreateFirewall{}
```

Comment:
nit, this is not necessary, first `append()` will initialize it.

Reply:
Deleted this line